pymc3_models/models/NaiveBayes.py

```diff
@@ -0,0 +1,283 @@
+
+        Parameters
+        ----------
+        X : numpy array, shape [num_training_samples, num_pred]. Contains the points
+            for which we want to predict the class
+
+        Returns
+        -------
+        A numpy array of shape [num_training_samples, num_cats] that contains the probabilities
+        that each sample belong to each category.
+
+        References
+        ----------
+        .. [1] Murphy, K. P. (2012). Machine learning: a probabilistic perspective.
+        """
+
+        if self.trace is None:
+            raise PyMC3ModelsError("Run fit() on the model before predict()")
+
+        predictions = np.array([])
+        for x in X:
+            prob_per_sample = scipy.stats.norm(self.trace['mu'], self.trace['sigma']).pdf(x)
+            prob_per_feature = [np.sum(prob_per_sample[:, :, i], axis=0)/len(self.trace['mu'])
+                                for i in range(self.num_pred)]
+            prob_per_class = normalize(ft.reduce(lambda x, y: x*y, prob_per_feature))
+            if len(predictions) == 0:
+                predictions = prob_per_class
+            else:
+                predictions = np.vstack((predictions, prob_per_class))
+
+        return predictions
+
+    def predict(self, X):
+        """
+        Classify new data with a trained Naive Bayes model.
+
+        Parameters
+        ----------
+        X : numpy array, shape [num_training_samples, num_pred]. Contains the data
+            to classify.
+
+        Returns
+        -------
+        A numpy array of shape [num_training_samples,] that contains the predicted class to
+        which the data points belong.
+        """
+        proba = self.predict_proba(X)
+        pred = np.argmax(proba, axis=1)
+        return pred
+
+    def score(self, X, y):
+        """
+        Scores new data with a trained model.
+
+        Parameters
+        ----------
+        X : numpy array, shape [num_training_samples, num_pred]. Contains the data points.
+
+        y : numpy array, shape [num_training_samples,]. Contains the category of the data points.
+
+        Returns
+        -------
+        A float representing the score of the predictions.
+        """
+
+        return accuracy_score(y, self.predict(X))
+
+    def save(self, file_prefix):
+        params = {
+            'inference_type': self.inference_type,
+            'num_cats': self.num_cats,
+            'num_pred': self.num_pred,
+            'num_training_samples': self.num_training_samples
+        }
+        super(GaussianNaiveBayes, self).save(file_prefix, params)
+
+    def load(self, file_profile):
+        params = super(GaussianNaiveBayes, self).load(file_profile, load_custom_params=True)
+
+        self.inference_type = params['inference_type']
+        self.num_cats = params['num_cats']
+        self.num_pred = params['num_pred']
+        self.num_training_samples = params['num_training_samples']
+
+
+# Functions
```

Comment:
Let's move this to another file called something like pymc_models/common/utils.py. That way it will be available to any other models that might need this.